src/main/java/uk/gov/hmcts/cmc/claimstore/services/pilotcourt/PilotCourtService.java

```diff
@@ -0,0 +1,115 @@
+package uk.gov.hmcts.cmc.claimstore.services.pilotcourt;
+
+import feign.FeignException;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.cmc.claimstore.courtfinder.CourtFinderApi;
+import uk.gov.hmcts.cmc.claimstore.services.ccd.legaladvisor.HearingCourt;
+import uk.gov.hmcts.cmc.claimstore.services.ccd.legaladvisor.HearingCourtMapper;
+import uk.gov.hmcts.cmc.claimstore.utils.ResourceReader;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import javax.annotation.PostConstruct;
+
+@Service
+public class PilotCourtService {
+
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    private Map<String, PilotCourt> pilotCourts;
+    private final CourtFinderApi courtFinderApi;
+    private final HearingCourtMapper hearingCourtMapper;
+    private final String dataSource;
+
+    public PilotCourtService(@Value("${pilot-courts.datafile}") String dataSource,
+                             CourtFinderApi courtFinderApi,
+                             HearingCourtMapper hearingCourtMapper) {
+        this.dataSource = dataSource;
+        this.hearingCourtMapper = hearingCourtMapper;
+        this.courtFinderApi = courtFinderApi;
+    }
+
+    @PostConstruct
+    public void init() {
+
+        try {
+            String data = ResourceReader.readString(dataSource);
+            CSVParser parse = CSVParser.parse(data, CSVFormat.DEFAULT);
+            Iterator<CSVRecord> iterator = parse.iterator();
+
+            Map<String, PilotCourt> pilotCourts = buildPilotCourts(iterator);
```

Comment:
I would propose that `init()` doesn't need to know how `buildPilotCourts()` will inspect the parse results; prefer `buildPilotCourts(CSVParser)` (or even accepting the raw data `String`)